File: Linked_List.java
class Node{
    int value;
    int key;
    Node next;
    public Node(){}
    public Node(int key, int value){
        this.value = value;
        this.key = key;
        this.next = null;
    }
    public boolean if_next(){
        return this.next != null;
    }
    public Node get_next(){
        return this.next;
    }
    public int get_value(){
        return this.value;
    }
    public void set_next(Node node){
        this.next = node;
    }
}

public class Linked_List{
    Node node_head = null;
    public Linked_List(){}
    public void add_last(int key, int value){
        Node new_node = new Node(key, value);
        if(node_head == null){
            node_head = new Node(key, value);
        }
        else{
            Node var = node_head;
            while(var.if_next()){
                var = var.get_next();
            }
            var.set_next(new_node);
        }
    }
    public void print_list(){
        Node var = node_head;
        System.out.print("[");
        while(var != null){
            System.out.print(" " + var.get_value());
            var = var.get_next();
        }
        System.out.println(" ]");
    }
    public void delete(int value){
        Node current = node_head;
        Node previous = null;
        while(current != null){
            if(current.get_value() == value){
                previous.set_next(current.get_next());
            }
            previous = current;
            current = current.get_next();
        }
    }
}

File: Linked_List.py
class Linked_List:
    class Node:
        def __init__(self, key, value):
            self._value = value
            self._key = key
            self._next = None

    def __init__(self):
        self.node_head = None

    def add_last(self, key, value):
        if not self.node_head:
            self.node_head = self.Node(key, value)
        else:
            last_var = self.node_head
            while last_var._next:
                last_var = last_var._next
            last_var._next = self.Node(key, value)

    def search(self, value):
        var = self.node_head
        while var:
            if var._key == value:
                return var
            var = var._next

    def delete(self, value):
        if self.node_head._value == value:
            self.node_head = self.node_head._next
        else:
            current = self.node_head
            previous = None
            while current:
                if current._value == value:
                    previous._next = current._next
                    return current
                previous = current
                current = current._next

    def __repr__(self):
        prnt = "["
        var = self.node_head
        while var:
            prnt += " " + var._value + ","
            var = var._next
        prnt = prnt.rstrip(",") + " ]"
        return prnt
